### DISCUSSION TITLE
Energy dashboard

### DISCUSSION THREAD

**Discussion by jaishkp** [wills106/homeassistant-solax-modbus#184] [General]
Got my solax gen4 x1 up and running this morning with a waveshare device.
All the outputs appear to be functioning correctly but I don’t seem to be able to setup the energy dashboard in home assistant correctly. Can someone please advise which sensor should be selected for each section? My flow diagram looks to be showing accumulative amount for the day rather than the live data.

**Comment by wills106** [comment]
Your Dashboard settings should be something like:

![image](https://user-images.githubusercontent.com/18155231/200118632-2b07a0f5-83ee-49b6-8093-87966a71d11c.png)

Just so you know the Energy Dashboard isn't actually live. It lags behind by an hour.

**Reply by wills106** [reply]
I have added more info in the [wiki](https://github.com/wills106/homeassistant-solax-modbus/wiki/Energy-Dashboard's)

**Reply by jaishkp** [reply]
Thank you!

**Reply by Zakarumit** [reply]
Hi,
 I have added Solax X3 G4 to HA without any issues, but when trying to setup the Energy dashboard, I cant pick any sensors it seems - I enclose a screenshot. Anyone could tell me whats the problem? I am a beginner with HA. Is it possible to create a dashboard with real time energy flows somehow?
![dashboard](https://user-images.githubusercontent.com/121687948/211199148-6af66e3a-b023-4103-8b4c-5944d67b988e.jpg)

**Reply by wills106** [reply]
On the integration page how entities does the SolaX Integration show?
Does it just show 4 or over 100?

**Reply by Zakarumit** [reply]
1 device / 208 entities. Everything looks good here, on overview I can see them all as well with values already. Is it perhaps necessary to wait some time after the installation until the dashboard configurator can see the entities? But its over an hour and still it cant see them it seems.

**Reply by Zakarumit** [reply]
...from those 208 entities part is disabled by integration, so it could be about those 100 working that you mentioned give or take

**Reply by wills106** [reply]
I couldn't remember how many the Gen4 generates.
I just wanted to make sure the Integration was working correctly.

If you look at wiki there is info on a real time card you can use. (There is a direct link a couple of posts up)

If you start typing SolaX or whatever you named the Instance, does anything start to display then to select?

**Reply by Zakarumit** [reply]
Nope, when I try typing SolaX, it doesnt display anything to select. On overview I see it all though, like the first needed for the Energy Dashboard: SolaX Today's Import Energy saying something like 10.33 kWh right now. Everything seems to work but the Energy dashboard selection.

**Reply by Zakarumit** [reply]
Hi, after the update I can see the sensors in Energy now. Thanks

**Comment by solarjones** [comment]
I'm assuming those options are only available for SolaX as I don't seem those with the Solis_Hybrid ?  Only has totals which aren't correct for the energy dashboard but at an entity level the data is there. Not sure how I expose them to the dashboard.

**Reply by wills106** [reply]
On the Solis you should have:

"Power Generation Today"
"Grid Import Today"
"Grid Export Today"
"Battery Charge Today"
"Battery Discharge Today"

**Reply by solarjones** [reply]
All others exist when assigning to the Energy dashboard but not Grid Import Today or Grid Export Today. Instead it lists:
- Meter Grid Import Total
- Meter Grid Export Total

<img width="423" alt="image" src="https://user-images.githubusercontent.com/100320468/201500072-971891f2-8d5a-46fa-8adf-229a31958bf2.png">


I can see them in the Device Info sensors but they are hidden.
<img width="218" alt="image" src="https://user-images.githubusercontent.com/100320468/201500030-61c57c33-5b54-406f-a0c1-4a96fb57dc45.png">

How do they get exposed to the Energy dashboard? 
Thanks again!

**Reply by wills106** [reply]
If you click on the sensor, then advanced. You can enable the sensor. It takes 30 or so seconds for it to start responding.

In the next release I'll change Import / Export Today to not being disabled by default.

**Reply by jaygit81** [reply]
I have just setup my solis inverter and seems to be working good.
I am having trouble getting the energy dashboard to display the the correct data.
I have used as suggested:
"Power Generation Today"
"Grid Import Today"
"Grid Export Today"
"Battery Charge Today"
"Battery Discharge Today"
But this literally shows the daily totals in the energy distribution, not the live figures.
Any help much appreciated.

**Reply by wills106** [reply]
The built in Energy Dashboard isn't real time and only updates once an hour.
If you look in the Wiki there are examples of a live one.

**Reply by wills106** [reply]
@jaygit81 2022.12.7 adds real time Battery Input / Output sensors for the Solis Inverters.

**Reply by jaygit81** [reply]
@wills106 you read my mind.... that was what I was going to ask about next. Thanks for the great work on this.
The only thing that I can see that is missing now is "Alarms/Faults" It would be great to get a log of any faults from the inverter, similar to the solis cloud app. Is this even possible? Thanks again.

**Reply by wills106** [reply]
It's on the [Todo List V2](https://github.com/wills106/homeassistant-solax-modbus/discussions/200#discussioncomment-4242410)

**Comment by hallidaydaniel** [comment]
I have used them successfully, although my setup is a bit complex as I have an X1-AC attatched to a battery and then twp x1-BOOST attatched to two solar arrays. You can get everything you need through the combination: 
![image](https://user-images.githubusercontent.com/7305486/204662424-00673426-fb18-45fb-a329-25c289727b89.png)

Remember if you need to get a slightly unusual reading for something you can always create a template sensor in HA for it. I have found this useful for having some 'instantanious' power readings, alongside the energy dashboard. 
I have done this for the battery in/ out which just uses the battery charge power sensor from the integration, but present it always as a positive and across two different sensors, like this:

```

      - state: >
          {% if (states('sensor.solax_battery_power_charge') | float) >= 0 %}
            {{ (states('sensor.solax_battery_power_charge') | float) }}
          {% else %}
            {{ 0 }}
          {% endif %}
        name: "Solax battery power in"
        unit_of_measurement: "W"
        device_class: power
        state_class: measurement

      - state: >
          {% if (states('sensor.solax_battery_power_charge') | float) < 0 %}
            {{ (states('sensor.solax_battery_power_charge') | float) * -1 }}
          {% else %}
            {{ 0 }}
          {% endif %}
        name: "Solax battery power out"
        unit_of_measurement: "W"
        device_class: power
        state_class: measurement
```
